src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.ReadObjectResponse.liquid

```diff
@@ -0,0 +1,33 @@
+﻿protected virtual async System.Threading.Tasks.Task<ObjectResponseResult<T>> ReadObjectResponseAsync<T>(System.Net.Http.HttpResponseMessage response, System.Collections.Generic.IReadOnlyDictionary<string, System.Collections.Generic.IEnumerable<string>> headers)
+{
+#if NSWAG_STRING_RESPONSES // Define this symbol in your project properties to enable inspecting response bodies as strings.
```

Comment:
Maybe we should generate a property on the client instead of this compiler switch? This way you can use it in code also if its published eg via nuget...

The problem with this is that you will probably the only one who knows about this/uses this otherwise

Reply:
I like that idea. But there are a few options:

* Single static class `NwagOptions` with a static `Boolean ExposeStringResponses` property.
    * Advantages: Only need to set a single property to enable inspection of all responses regardless of generated client class or when using multiple NSwag-generated libraries.
* Static `Boolean ExposeStringResponses` property on each generated Client class.
     * Advantages: More granularity.
* Instance `Boolean ExposeStringResponses` property on each generated Client class.
     * Advantages: Even greater granularity - but complicates the interface. Mock/testing classes would need to be updated to include this property and alter their behaviour too.
* As a method parameter with a default value, e.g. `GetSomething( String param1, Boolean exposeStringResponses = false, CancellationToken ct = default );`
    * Advantages: Avoids race-conditions.
* Private field set in the `Client` constructor or factory method.
    * Advantages: Doesn't change interface, avoids race-conditions. I'm unsure if consumers would want to change the string-exposure behaviour after the Client object has been created. Presumably it's a setting they'd keep in their `appSettings.json` to help investigate runtime exceptions from unexpected responses.

Reply:
I'd go with 

> Instance read/write Boolean ExposeStringResponses property on each generated Client class.

as static properties are bad for isolation between different users in the same process.

The property should only be on the class itself not on the client interface...

We could still make the default of the property dependent on a compile switch, e.g. when NSWAG_STRING_RESPONSES is set then set the property to true otherwise false.